### ISSUE TITLE
Feature Request: New function that returns covariance matrix

### ISSUE THREAD

**Issue by mdancho84** [tidymodels/corrr#42]
Great package. Something to consider is a function similar to `correlate()` that would return the covariance matrix. This will greatly assist with financial analysis.

**Comment by drsimonj** [comment]
After some thinking, I have some questions:

- If a separate function, what would you call it? E.g., for correlations we have `correlate()`, for covariance we have ... ?
- Regardless, would a separate function be best? An alternative is to include a new argument in `correlate()`. E.g., could be a boolean like `standardise` that is `TRUE` by default for calculating correlations, but if `FALSE` computes covariance.
- In general, how would you like the diagonal to be handled? `correlate()` returns whole diagonal as missing, and a `diagonal` argument allows them to all be set to anything such as 1, which is typical of correlation matrix. Covariance matrices typically have variance on the diagonal which is not uniform. If a new function is created, perhaps something like a boolean `diag_as_na` argument is used that returns diagonal as missing if `TRUE` or variance otherwise.
- `correlate()` returns a tibble with a `cor_df` class. Would there be any reason to change this for a covariance structure? I.e., are there any other functions in corrr that you would want to behave differently with a covariance matrix? I can think of two: `rplot` and `network_plot` make colours range from -1 to 1 by default.

**Comment by mdancho84** [comment]
First, I think this will be a very good addition to your package. From the financial side, the intended use will be with stock returns over time to understand how multiple stocks move together. This article explains more about how I will end up using it. http://www.investopedia.com/articles/financial-theory/11/calculating-covariance.asp

I'll collect some feedback from others as well, but here are some initial thoughts:

- Function name is a tough one. `covariate()` might work since it follows the verb philosophy of the "tidyverse". Although `covariation()` is a noun, it might work well since users will know what it's doing. My preference is covariate. 
-  A separate function makes sense to me since base R already has two separate functions `cor()` and `cov()`. However, I can see your point since the two are related. I actually like the `standardise` (and hopefully `standardize`) argument since essentially that is the essence. I would definitely support this approach.
- The diagonal would need to be the variance for the covariance data frame. I'm ok with shaving off the top or bottom, but the diagonal should be present.  
- I think you can keep the `cor_df` class. I would not change this unless there is a specific reason to. However, you may have identified with `rplot` and `network_plot` colors. I will investigate more.

Overall, I think the `correlate(standarise = TRUE)` could be a nice solution. However, we'll need to investigate the interaction with the `rplot` and `network_plot`. Otherwise it may make sense to do a separate function with separate classes.

**Comment by drsimonj** [comment]
I like `covariate()`. I think it's a better option.

**Comment by mdancho84** [comment]
I'm on board with that. It seems that it will be easier to handle for `rplot` and `network_plot`.

**Comment by edgararuiz** [comment]
Is this request still current?

**Comment by juliasilge** [comment]
Closed in #116 thanks to @jameslairdsmith:

``` r
library(corrr)
colpair_map(mtcars, cov)
#> # A tibble: 11 x 12
#>    rowname     mpg     cyl   disp      hp     drat      wt     qsec       vs
#>    <chr>     <dbl>   <dbl>  <dbl>   <dbl>    <dbl>   <dbl>    <dbl>    <dbl>
#>  1 mpg       NA     -9.17  -633.  -321.     2.20    -5.12    4.51     2.02  
#>  2 cyl       -9.17  NA      200.   102.    -0.668    1.37   -1.89    -0.730 
#>  3 disp    -633.   200.      NA   6721.   -47.1    108.    -96.1    -44.4   
#>  4 hp      -321.   102.    6721.    NA    -16.5     44.2   -86.8    -25.0   
#>  5 drat       2.20  -0.668  -47.1  -16.5   NA       -0.373   0.0871   0.119 
#>  6 wt        -5.12   1.37   108.    44.2   -0.373   NA      -0.305   -0.274 
#>  7 qsec       4.51  -1.89   -96.1  -86.8    0.0871  -0.305  NA        0.671 
#>  8 vs         2.02  -0.730  -44.4  -25.0    0.119   -0.274   0.671   NA     
#>  9 am         1.80  -0.466  -36.6   -8.32   0.190   -0.338  -0.205    0.0423
#> 10 gear       2.14  -0.649  -50.8   -6.36   0.276   -0.421  -0.280    0.0766
#> 11 carb      -5.36   1.52    79.1   83.0   -0.0784   0.676  -1.89    -0.464 
#> # … with 3 more variables: am <dbl>, gear <dbl>, carb <dbl>
```

<sup>Created on 2020-11-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>
